AWSPinpointTests/AWSPinpointTargetingClientTests.m

```diff
@@ -249,49 +241,56 @@ - (void)testNewDeviceTokenUpdateWithStandardUserDefaults {
     [userDefaults synchronize];
 }
 
-- (void)testCurrentProfileWithSystemOptOutAndApplicationOptOut {
-    [self setSystemLevelOptOut:YES];
-    [self setApplicationLevelOptOut:YES];
-
+- (void)testCurrentProfileReturnsOptOutAll {
+    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:NO] forceCreate:YES];
     AWSPinpointEndpointProfile *profile = [self.pinpoint.targetingClient currentEndpointProfile];
     XCTAssertTrue([profile.optOut isEqualToString:@"ALL"]);
 }
 
-- (void)testCurrentProfileWithSystemOptOutAndApplicationOptIn {
-    [self setSystemLevelOptOut:YES];
-    [self setApplicationLevelOptOut:NO];
+- (void)testCurrentProfileReturnsOptOutAllWhenNotificationsEnabledAndDeviceTokenNotSet {
+    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:NO] forceCreate:YES];
+    [self initializeMockApplicationWithRemoteNotifications:YES];
 
     AWSPinpointEndpointProfile *profile = [self.pinpoint.targetingClient currentEndpointProfile];
-
     XCTAssertTrue([profile.optOut isEqualToString:@"ALL"]);
 }
 
-- (void)testCurrentProfileWithSystemOptOutAndApplicationOptOutUnset {
-    [self setSystemLevelOptOut:YES];
+- (void)testCurrentProfileReturnsOptOutAllWhenNotificationsDisabledAndDeviceTokenSet {
+    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:NO] forceCreate:YES];
+    [self setDeviceTokenInUserDefaults];
 
     AWSPinpointEndpointProfile *profile = [self.pinpoint.targetingClient currentEndpointProfile];
-
     XCTAssertTrue([profile.optOut isEqualToString:@"ALL"]);
 }
 
-- (void)testCurrentProfileWithSystemOptInAndApplicationOptOut {
-    [self setSystemLevelOptOut:NO];
-    [self setApplicationLevelOptOut:YES];
+- (void)testCurrentProfileReturnsOptOutNoneWhenNotificationsEnabledAndDeviceTokenSet {
+    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:NO] forceCreate:YES];
+    [self initializeMockApplicationWithRemoteNotifications:YES];
+    [self setDeviceTokenInUserDefaults];
 
     AWSPinpointEndpointProfile *profile = [self.pinpoint.targetingClient currentEndpointProfile];
+    XCTAssertTrue([profile.optOut isEqualToString:@"NONE"]);
+}
 
+- (void)testCurrentProfileReturnsOptOutAllForApplicationLevelOptOut {
+    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:YES] forceCreate:YES];
+    [self initializeMockApplicationWithRemoteNotifications:YES];
+    [self setDeviceTokenInUserDefaults];
+
+    AWSPinpointEndpointProfile *profile = [self.pinpoint.targetingClient currentEndpointProfile];
     XCTAssertTrue([profile.optOut isEqualToString:@"ALL"]);
 }
 
-- (void) testCurrentProfileWithSystemOptInAndApplicationOptOutBackgroundThread {
+- (void) testCurrentProfileWithNotificationsEnabledBackgroundThread {
```

Comment:
should we have a similar test where we notification is disabled by not setting the deviceTokenInUserDefaults?

Reply:
ah lost my comment, this should be covered by `testCurrentProfileReturnsOptOutAllWhenNotificationsEnabledAndDeviceTokenNotSet`, but will add this for the execution on background thread testing